packages/devkit/src/utils/get-workspace-layout.ts

```diff
@@ -36,6 +36,24 @@ export function getWorkspaceLayout(tree: Tree): {
   };
 }
 
+export function extractLayoutDirectory(directory: string): {
+  layoutDirectory: string;
+  projectDirectory: string;
+} {
+  if (directory) {
+    directory = directory.startsWith('/') ? directory.substring(1) : directory;
+    for (let dir of ['apps', 'libs', 'packages']) {
```

Comment:
I believe it would make things simpler if this included the dirs from `workspaceLayout`?

```ts
const { appsDir, libsDir } = getWorkspaceLayout(tree); 
let dir of ['apps', 'libs', 'packages', appsDir, libsDir] {
```

I think you wouldn't have to return the `layoutDirectory` as its almost always used to compare to the `workspaceLayout`.

Reply:
You will still have to return layourDirectory cause otherwise getWorkspaceLayout will return '.'

But I agree about including appsDir and libsDir

Reply:
> I think you wouldn't have to return the layoutDirectory as its almost always used to compare to the workspaceLayout.

The two functions can be combined into one (layoutDirectory would still have to be returned though). I thought about it but since the plan is to remove the appsDir and libsDir notions, I decided against it, so the change is smaller.